apollo-federation/src/error/mod.rs

```diff
@@ -13,6 +13,37 @@ use lazy_static::lazy_static;
 
 use crate::subgraph::spec::FederationSpecError;
 
+/// Break out of the current function, returning an internal error.
+#[macro_export]
+macro_rules! internal_error {
+    ( $( $arg:tt )+ ) => {
+        return Err($crate::error::FederationError::internal(format!( $( $arg )+ )).into());
```

Comment:
What is this `into` for? What are we casting `FederationError` into?

Reply:
It's to make it work like the `?` operator, but yeah, in practice that doesn't happen.

Reply:
Gotcha. I wasn't sure if there was a specific use.